src/asynchronous/embassy/mod.rs

```diff
@@ -316,11 +328,11 @@ impl<'a, M: RawMutex, B: I2c> Tps6699x<'a, M, B> {
 
     /// Trigger an `ANeg` command to autonegotiate the sink contract.
     pub async fn autonegotiate_sink(&mut self, port: PortId) -> Result<(), Error<B::Error>> {
-        self.execute_command(port, Command::Aneg, None, None)
-            .await?
-            .success_or(PdError::Failed)?;
-
-        Ok(())
+        match self.execute_command(port, Command::Aneg, None, None).await? {
+            ReturnValue::Success => Ok(()),
+            ReturnValue::Rejected => PdError::Rejected.into(),
```

Comment:
Personally, I find the `.into()` that becomes a `Result` confusing and would always prefer an explicit `Err(_)`